### PR TITLE
Show snackbars with duration indefinite. Add a button to dismiss them

### PR DESCRIPTION
**Pull request by ferazoguerrero**
## Ticket
- [#####]

## Github Issue
- ISSUE-

## Description
- Show snackbars with duration indefinite. 
- Add a button to dismiss them

## Commit message
- Improve snackbars in sample app

## Testing
- [ ] Added unit tests
- [x] Ran unit tests successfully
- [ ] Added documentation for public APIs and/or Wiki

### CODE REVIEW

File: sample/src/main/java/com/twilio/verify/sample/view/Extensions.kt [hunk 1]
```diff
@@ -34,7 +34,11 @@ fun Throwable.showError(
   (cause as? InputException)?.let {
     message = "$message. ${it.message}"
   }
-  Snackbar.make(content, message, BaseTransientBottomBar.LENGTH_LONG).show()
+  val snackbar = Snackbar.make(content, message, Snackbar.LENGTH_INDEFINITE)
+  snackbar.setAction(R.string.dismiss){
```
Comment (AuthyApps) [inline sample/src/main/java/com/twilio/verify/sample/view/Extensions.kt:38]: <table data-meta="generated_by_Linter">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-android/blob/1d551b5cdada7a5b7686b0c63d4e00df570174db/sample/src/main/java/com/twilio/verify/sample/view/Extensions.kt#L38"></span>Missing spacing before "{"</td>
    </tr>
  </tbody>
</table>

File: sample/src/main/java/com/twilio/verify/sample/view/factors/create/CreateFactorFragment.kt [hunk 2]
```diff
@@ -148,12 +147,15 @@ class CreateFactorFragment : Fragment() {
...
     (exception.cause as? NetworkException)?.failureResponse?.apiError?.let {
-      Snackbar.make(
+      val snackbar = Snackbar.make(
         content,
         "Code: ${it.code} - ${it.message}",
-        BaseTransientBottomBar.LENGTH_LONG
+        BaseTransientBottomBar.LENGTH_INDEFINITE
       )
-        .show()
+      snackbar.setAction(R.string.dismiss){
```
Comment (AuthyApps) [inline sample/src/main/java/com/twilio/verify/sample/view/factors/create/CreateFactorFragment.kt:155]: <table data-meta="generated_by_Linter">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-android/blob/1d551b5cdada7a5b7686b0c63d4e00df570174db/sample/src/main/java/com/twilio/verify/sample/view/factors/create/CreateFactorFragment.kt#L155"></span>Missing spacing before "{"</td>
    </tr>
  </tbody>
</table>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Fix code convention problems</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request title does not follow the convention [00000] PR Title text</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request title does not follow the convention [00000] PR Title text</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request title does not follow the convention [00000] PR Title text</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: sample/src/main/java/com/twilio/verify/sample/AppApplication.kt [hunk 3]
```diff
@@ -50,7 +51,7 @@ class AppApplication : Application() {
     super.onCreate()
     FirebaseApp.initializeApp(this)
     startKoin {
-      androidLogger()
+      androidLogger(level = Level.ERROR)
```
Comment (yafuquen) [inline sample/src/main/java/com/twilio/verify/sample/AppApplication.kt:54]: Why are we changing this?

File: sample/src/main/java/com/twilio/verify/sample/view/factors/create/CreateFactorFragment.kt [hunk 4]
```diff
@@ -148,12 +147,15 @@ class CreateFactorFragment : Fragment() {
 
   private fun handleNetworkException(exception: TwilioVerifyException) {
     (exception.cause as? NetworkException)?.failureResponse?.apiError?.let {
-      Snackbar.make(
+      val snackbar = Snackbar.make(
         content,
         "Code: ${it.code} - ${it.message}",
-        BaseTransientBottomBar.LENGTH_LONG
+        BaseTransientBottomBar.LENGTH_INDEFINITE
```
Comment (yafuquen) [inline sample/src/main/java/com/twilio/verify/sample/view/factors/create/CreateFactorFragment.kt:153]: Should it be `Snackbar.LENGTH_INDEFINITE`?

File: sample/src/main/java/com/twilio/verify/sample/AppApplication.kt (hunk 3 above)
Comment (ferazoguerrero) [reply sample/src/main/java/com/twilio/verify/sample/AppApplication.kt]: it was causing an error in my building on main branch, but it didn't in dev. Removed

File: sample/src/main/java/com/twilio/verify/sample/view/factors/create/CreateFactorFragment.kt (hunk 4 above)
Comment (ferazoguerrero) [reply sample/src/main/java/com/twilio/verify/sample/view/factors/create/CreateFactorFragment.kt]: Changed

**Pull request merged**